spec/double_entry/reporting/line_aggregate_filter_spec.rb

```diff
@@ -0,0 +1,96 @@
+# encoding: utf-8
+RSpec.describe DoubleEntry::Reporting::LineAggregateFilter do
+  describe '.filter' do
+    let(:function) { :sum }
+    let(:account) { :account }
+    let(:code) { :transfer_code }
+    let(:filter_criteria) { nil }
+    let(:start) { Time.parse('2014-07-27 10:55:44 +1000') }
+    let(:finish) { Time.parse('2015-07-27 10:55:44 +1000') }
+    let(:range) do
+      instance_double(DoubleEntry::Reporting::MonthRange,
+        :start => start,
+        :finish => finish,
+      )
+    end
+
+    let(:lines_scope) { spy(DoubleEntry::Line) }
+
+    subject(:filter) do
+      DoubleEntry::Reporting::LineAggregateFilter.new(
+        account, code, range, filter_criteria
+      )
+    end
+
+    before do
+      stub_const('DoubleEntry::Line', lines_scope)
+
+      allow(DoubleEntry::LineMetadata).to receive(:table_name).and_return("double_entry_line_metadata")
+
+      allow(lines_scope).to receive(:where).and_return(lines_scope)
+      allow(lines_scope).to receive(:joins).and_return(lines_scope)
+      allow(lines_scope).to receive(:ten_dollar_purchases).and_return(lines_scope)
+      allow(lines_scope).to receive(:ten_dollar_purchases_by_category).and_return(lines_scope)
+
+      filter.filter
+    end
+
+    context 'with named scopes specified' do
+      let(:filter_criteria) do
+        [
+          # an example of calling a named scope called with arguments
+          {
+            :scope => {
+              :name => :ten_dollar_purchases_by_category,
+              :arguments => [:cat_videos, :cat_pictures]
+            }
+          },
+          # an example of calling a named scope with no arguments
+          {
+            :scope => {
+              :name => :ten_dollar_purchases
+            }
+          },
+          # an example of providing a single metadatum criteria to filter on
```

Comment:
yup. its kinda :hankey:. Its a non trivial task to support multiple metadata filters on a single aggregate. How do you join the tables, and query? Does it mean `AND` or does it mean `OR`?

Considering the stuff we need it for now, we only need to support a single metadata filter per aggregate. We can solve this in another PR, and keep this one as small as possible